src/SwatchWithForm.tsx

```diff
@@ -1,16 +1,22 @@
 import Swatch from './Swatch';
 import Form from './Form';
 import { SwatchConfig } from './types'
+import { useState } from "react";
 
 function SwatchWithForm({swatchConfig, setSwatchConfig}  : { swatchConfig: SwatchConfig, setSwatchConfig: (arg0: SwatchConfig) => void}) {
+  const [displayData, setDisplayData] = useState({
+    showRowNumbers: false
+  })
```

Comment:
This is already in the `SwatchWithForm`; what if we just put the `showRowNumbers` checkbox directly at this level (between `Form` and `Swatch`) instead of nesting it in the `Form`?

It actually doesn't even _need_ to be stateful, though I imagine you like having it included in the url params with the rest of the state

Reply:
Having it at this level would mean you could get rid of the case-based evaluation of `setFormValue` that you dislike

Reply:
It's not in the url anyway, I dont' want it to be stateful haha. Yeah that was my thought, if we don't go this way we have an extra display form (display stuff doesn't end up on the URL, and it's row numbers, orientation of the thing, and mirroring if I decide to do that)